core/src/main/java/org/elasticsearch/action/admin/cluster/settings/TransportClusterUpdateSettingsAction.java

```diff
@@ -148,7 +149,7 @@ public void onNoLongerMaster(String source) {
                     @Override
                     public void onFailure(String source, Exception e) {
                         //if the reroute fails we only log
-                        logger.debug("failed to perform [{}]", e, source);
+                        logger.debug(new ParameterizedMessage("failed to perform [{}]", source), e);
```

Comment:
It's only needed for the invocations passing a throwable, because of the overrides that are available in the Log4j 2 API. That said, I used a script for converting all invocations of `logger.(error|warn|info|debug|trace)(String, Throwable, Object...) -> logger.$1(new ParameterizedMessage(String), Object...), Throwable)` and a second script for converting all `new ParameterizedMessage(String, Object...) -> `(Supplier<?>) () -> new ParameterizedMessage(String, Object)` that I was using to keep my branch up to date with master, and I forgot to run the script before prepping this PR (it requires manual edits for line length violations after running). I will update this PR for that once I get home.